lib/db.js

```diff
@@ -150,9 +150,11 @@ class LimitDBRedis extends EventEmitter {
     const bucket = this.buckets[params.type];
     const bucketKeyConfig = this.bucketKeyConfig(bucket, params);
 
-    let count = params.count || 1;
-    if (count === 'all') {
+    let count = 1;
+    if (params.count === 'all') {
       count = bucketKeyConfig.size;
+    } else if (Number.isInteger(params.count)) {
+      count = params.count;
     }
```

Comment:
Could we fail if not an integer/all/nil? I assume no one is passing count `0.5`/`"1"`, but still, better to fail if we are going to ignore a non-nil argument.